public/hooks/__tests__/use_sessions.test.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { useDeleteSession, usePatchSession } from '../use_sessions';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useCore } from '../../contexts/core_context';
+import { HttpHandler } from '../../../../../src/core/public';
+
+jest.mock('../../contexts/core_context');
+const useCoreMocked = useCore as jest.MockedFunction<typeof useCore>;
+
+describe('useDeleteSession', () => {
+  it('should call delete with path and signal', async () => {
+    const { result } = renderHook(() => useDeleteSession());
+
+    await act(async () => {
+      await result.current.deleteSession('foo');
+    });
+    expect(useCoreMocked.mock.results[0].value.services.http.delete).toHaveBeenCalledWith(
+      '/api/assistant/session/foo',
+      expect.objectContaining({
+        signal: expect.any(Object),
+      })
+    );
+  });
+
+  it('should be loading after deleteSession called', async () => {
+    const { result, waitFor } = renderHook(() => useDeleteSession());
+    useCoreMocked.mock.results[0].value.services.http.delete.mockReturnValue(new Promise(() => {}));
+
+    act(() => {
+      result.current.deleteSession('foo');
+    });
+
+    await waitFor(() => {
+      expect(result.current.loading).toBe(true);
+    });
+  });
+
+  it('should return data after delete success', async () => {
+    const { result, waitFor } = renderHook(() => useDeleteSession());
+    useCoreMocked.mock.results[0].value.services.http.delete.mockReturnValue(
+      Promise.resolve('deleted')
+    );
+
+    await act(async () => {
```

Comment:
Q: curious when we should use `await act`?

Reply:
The `await act` can be removed here. I think we may need `await act` for async callbacks.
If we remove `await act` and `await waitFor` in this test case, the console will output a warning like this:
![image](https://github.com/opensearch-project/dashboards-assistant/assets/4034161/9eefb71d-3c18-43ba-b155-f765eb32ed5c)

So I think we need add `await` prefix before `act` when ends with non-async actions.